x-pack/plugins/saved_objects_tagging/server/services/tags/tags_client.ts

```diff
@@ -63,6 +73,17 @@ export class TagsClient implements ITagsClient {
     return results.map(savedObjectToTag);
   }
 
+  public async findByColor(color: string) {
+    const queryVersion = this.soClient.getQueryVersion(this.type);
+    const useColorField = queryVersion < 4;
+
+    const tags = this.soClient.find({
+      type: this.type,
+      filter: useColorField ? `tag.color: ${color}` : `tag.colors: ${color}`,
+    });
+    return tags;
```

Comment:
Example of how using a different query depending on the model version could look like